src/django_components/component.py

```diff
@@ -1579,9 +2469,48 @@ def renderer(root_attributes: Optional[List[str]] = None) -> Tuple[str, Dict[str
 
         return renderer
 
+    def _call_data_methods(self, args: Any, kwargs: Any, slots: Any, context: Context) -> Tuple[Dict, Dict, Dict]:
+        # If user doesn't specify `Args`, `Kwargs`, `Slots` types, then we pass them in as plain
+        # dicts / lists.
+        args_inst = self.Args(*args) if self.Args is not None else args
```

Comment:
Here's the logic that does the trick that, when one defines `Component.Args`, so that the `get_template_data()` receives an instance of the `Args` class, instead of plain dicts / lists.